ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/toolAdmin/view/subviews/ToolVolumesFromListEditor.java

```diff
@@ -0,0 +1,221 @@
+        columns.add(tag);
+        columns.add(url);
+        columns.add(readOnly);
+        ColumnModel<ToolVolumesFrom> cm = new ColumnModel<ToolVolumesFrom>(columns);
+
+        grid = new Grid<ToolVolumesFrom>(listStore, cm);
+
+        editing = new GridInlineEditing<ToolVolumesFrom>(grid);
+        enableGridEditing(name, namePrefix, tag, url, readOnly);
+
+        initWidget(grid);
+    }
+
+    private void enableGridEditing(ColumnConfig<ToolVolumesFrom, String> name,
+                                   ColumnConfig<ToolVolumesFrom, String> namePrefix,
+                                   ColumnConfig<ToolVolumesFrom, String> tag,
+                                   ColumnConfig<ToolVolumesFrom, String> url,
+                                   ColumnConfig<ToolVolumesFrom, Boolean> readOnly) {
+        final TextField nameTextField = new TextField();
+        nameTextField.setAllowBlank(false);
+        editing.addEditor(name, nameTextField);
+
+        final TextField namePrefixTextfield = new TextField();
+        namePrefixTextfield.setAllowBlank(false);
+        editing.addEditor(namePrefix, namePrefixTextfield);
+
+        final TextField tagTextField = new TextField();
+        editing.addEditor(tag, tagTextField);
+
+        final TextField urlTextField = new TextField();
+        urlTextField.addValidator(new UrlValidator());
+        editing.addEditor(url, urlTextField);
+
+        final CheckBox readOnlyCheckBox = new CheckBox();
+        editing.addEditor(readOnly, readOnlyCheckBox);
+
+        editing.addCompleteEditHandler(new CompleteEditEvent.CompleteEditHandler<ToolVolumesFrom>() {
+            @Override
+            public void onCompleteEdit(CompleteEditEvent<ToolVolumesFrom> event) {
+                String name = nameTextField.getCurrentValue();
+                String namePrefix = namePrefixTextfield.getCurrentValue();
+                String tag = tagTextField.getCurrentValue();
+                String url = urlTextField.getCurrentValue();
+                Boolean readOnly = readOnlyCheckBox.getValue();
+                ToolVolumesFrom volumesFrom = listStore.get(0);
+                volumesFrom.setName(name);
+                volumesFrom.setNamePrefix(namePrefix);
+                volumesFrom.setTag(tag);
+                volumesFrom.setUrl(url);
+                volumesFrom.setReadOnly(readOnly);
+                listStore.update(volumesFrom);
+            }
+        });
+    }
+
+    @Override
+    public Editor<List<ToolVolumesFrom>> asEditor() {
+        return new ListStoreEditor<>(listStore);
+    }
+
+    public void addVolumesFrom() {
+        ToolAutoBeanFactory factory = GWT.create(ToolAutoBeanFactory.class);
+        ToolVolumesFrom volumesFrom = factory.getVolumesFrom().as();
+        getVolumesFromTag(volumesFrom);
+
+        volumesFrom.setName("");
+        volumesFrom.setUrl("");
+        volumesFrom.setTag("");
+        volumesFrom.setNamePrefix("");
+        volumesFrom.setReadOnly(true);
```

Comment:
For JSON models like this, it's safer to leave values `null` by default. The service may allow a field to be optional but may not allow blank values. So if the user doesn't fill in a value for fields like this, they should be left null so that when the model is serialized to JSON, the field will be omitted. Also, we generally don't want to save blank values in the database, so we should avoid submitting blank values from the UI (the services should validate a field isn't blank, but they don't always).